relay-server/src/actors/envelopes.rs

```diff
@@ -182,6 +182,91 @@ impl ProcessingError {
     }
 }
 
+/// Contains the required envelope related information to create an outcome
+#[derive(Clone, Copy)]
+pub struct EnvelopeContext {
+    envelope_summary: EnvelopeSummary,
+    timestamp: DateTime<Utc>,
+    event_id: Option<EventId>,
+    remote_addr: Option<net::IpAddr>,
+    scoping: Scoping,
+}
+
+impl EnvelopeContext {
+    pub fn new(
+        envelope_summary: EnvelopeSummary,
+        timestamp: DateTime<Utc>,
+        event_id: Option<EventId>,
+        remote_addr: Option<net::IpAddr>,
+        scoping: Scoping,
+    ) -> Self {
```

Comment:
At the moment the constructor takes: `event_id`, `remote_addr`, `envelope_summary`, `timestamp` and `scoping`.
As you point out `scoping` must be provided outside the context of the envelope, also `timestamp` is extracted from the request.
`event_id` and `remote_addr` can always be extracted from the envelope but `envelope_summary` would be a bit instabile. I use the envelope summary from the `ProcessEnvelopeState` which is the summary before processing. If I had to extract the summary from the envelope at various times it would change the current implementation.

I will not make this change until we can double check that this is indeed what we want.